sentry/src/event_aggregator.rs

```diff
@@ -0,0 +1,127 @@
+use crate::access::check_access;
+use crate::db::event_aggregate::insert_event_aggregate;
+use crate::db::DbPool;
+use crate::event_reducer;
+use crate::Application;
+use crate::ResponseError;
+use crate::Session;
+use async_std::sync::RwLock;
+use chrono::Utc;
+use primitives::adapter::Adapter;
+use primitives::sentry::{Event, EventAggregate};
+use primitives::{Channel, ChannelId};
+use slog::{error, Logger};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::delay_for;
+
+pub(crate) type Aggregate = Arc<RwLock<HashMap<ChannelId, EventAggregate>>>;
+
+#[derive(Default, Clone)]
+pub struct EventAggregator {
+    aggregate: Aggregate,
+}
+
+pub fn new_aggr(channel_id: &ChannelId) -> EventAggregate {
+    EventAggregate {
+        channel_id: channel_id.to_owned(),
+        created: Utc::now(),
+        events: HashMap::new(),
+    }
+}
+
+async fn store(db: &DbPool, channel_id: &ChannelId, logger: &Logger, aggr: Aggregate) {
```

Comment:
The naming (or something) is quite confusing here

in `record`, you get an `aggr` FROM the `recorder`, but here you somehow turn the aggr INTo a `recorder`? how does this work?